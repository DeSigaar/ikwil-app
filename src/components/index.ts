export { default as Backbutton } from './Backbutton'
export { default as SearchBar } from './SearchBar'
export { default as Modal } from './Modal'
export { default as Filter } from './Filter'
export { default as FilterItem } from './FilterItem'
export { default as Main } from './Main'
export { default as Activity } from './Activity'
export { default as Icon } from './Icon'
export { default as Loader } from './Loader'
export * from './toast'
export * from './header'
