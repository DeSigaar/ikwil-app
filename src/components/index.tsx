export { default as Header } from './Header'
export { default as SearchBar } from './SearchBar'
export { default as Modal } from './Modal'
export { default as BackButton } from './BackButton'
export { default as Main } from './Main'
