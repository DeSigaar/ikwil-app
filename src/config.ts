export const FIREBASE_CONFIG = {
	apiKey: "AIzaSyAFtXOSXoFHgOB9FMOkdrFeJdTEvyGmm6U",
	appId: "1:35006139161:web:64f0195a7d8b7b99619388",
	authDomain: "stichting-ik-wil.firebaseapp.com",
	databaseURL: "https://stichting-ik-wil.firebaseio.com",
	measurementId: "G-5CNKSKZ241",
	messagingSenderId: "35006139161",
	projectId: "stichting-ik-wil",
	storageBucket: "stichting-ik-wil.appspot.com",
};
