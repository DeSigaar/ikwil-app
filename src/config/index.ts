export { default as firebase } from './firebase'
export { default as firebaseUI } from './firebaseUI'
export { default as localForage } from './localForage'
export { default as reactReduxFirebase } from './reactReduxFirebase'
