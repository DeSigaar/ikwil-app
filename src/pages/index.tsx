export { default as App } from './app'
export { default as LogIn } from './log-in'
export { default as Settings } from './settings'
export { default as PrivacyPolicy } from './privacy-policy'
export { default as TermsOfService } from './terms-of-service'
export { default as NotFound } from './not-found'
