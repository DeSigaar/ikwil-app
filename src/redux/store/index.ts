import {
  applyMiddleware,
  compose,
  createStore,
  Store,
  combineReducers,
} from 'redux'
import thunk from 'redux-thunk'
import { routerMiddleware, connectRouter } from 'connected-react-router'
import { getFirebase, firebaseReducer } from 'react-redux-firebase'
import { firestoreReducer } from 'redux-firestore'
import history from 'src/utils/history'

// Import all reducers
import appReducer from 'src/redux/app/reducers'

const rootReducer = combineReducers({
  // Combine all reducers
  app: appReducer,

  // Below are other reducers
  router: connectRouter(history),
  firebase: firebaseReducer,
  firestore: firestoreReducer,
})

export type RootState = ReturnType<typeof rootReducer>

const configureStore = (initialState: object): Store => {
  const composeEnhancer: typeof compose =
    (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(
      applyMiddleware(
        ...[routerMiddleware(history), thunk.withExtraArgument(getFirebase)],
      ),
    ),
  )

  return store
}

const initialState = window && (window as any).__INITIAL_STATE__

const store = configureStore(initialState)

export default store
