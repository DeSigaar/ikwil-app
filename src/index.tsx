import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'connected-react-router'
import { Switch, Route, Redirect } from 'react-router'
import { ReactReduxFirebaseProvider } from 'react-redux-firebase'
import { createFirestoreInstance } from 'redux-firestore'
import { App, Login, PrivacyPolicy, TermsOfService, NotFound } from 'src/pages'
import * as serviceWorker from 'src/serviceWorker'
import * as setupEvents from 'src/setupEvents'
import history from 'src/utils/history'
import store from 'src/redux/store'
import { fireApp } from 'src/utils/firebase'
import configReactReduxFirebase from 'src/config/reactReduxFirebase'
import { createGlobalStyle } from 'styled-components'
import { colors, fonts, layout } from 'src/styles'

const GlobalStyle = createGlobalStyle`
*,
*:before,
*:after {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background-color: ${colors.colors.white};
  margin: ${layout.init.margin};
  padding-top: ${layout.init.padding};
  font-family: ${fonts.font.fontFamily};
}
`

ReactDOM.render(
  <Provider store={store}>
    <GlobalStyle />
    <ConnectedRouter history={history}>
      <ReactReduxFirebaseProvider
        firebase={fireApp}
        config={configReactReduxFirebase}
        dispatch={store.dispatch}
        createFirestoreInstance={createFirestoreInstance}
      >
        <Switch>
          {/* App */}
          <Route exact path="/" component={App} />

          {/* Login */}
          <Route exact path="/login" component={Login} />
          <Redirect from="/inlog" to="/login" />
          <Redirect from="/inloggen" to="/login" />
          <Redirect from="/register" to="/login" />
          <Redirect from="/registreren" to="/login" />

          {/* Privacy Policy */}
          <Route exact path="/privacy-policy" component={PrivacyPolicy} />
          <Redirect from="/privacy" to="/privacy-policy" />
          <Redirect from="/privacypolicy" to="/privacy-policy" />

          {/* Terms of Service */}
          <Route exact path="/terms-of-service" component={TermsOfService} />
          <Redirect from="/tos" to="/terms-of-service" />
          <Redirect from="/terms" to="/terms-of-service" />
          <Redirect from="/termsofservice" to="/terms-of-service" />

          {/* 404 - No route found */}
          <Route component={NotFound} />
        </Switch>
      </ReactReduxFirebaseProvider>
    </ConnectedRouter>
  </Provider>,
  document.getElementById('root'),
)

serviceWorker.register({
  updateCacheStatus: setupEvents.changeCacheStatus,
})
setupEvents.init()
