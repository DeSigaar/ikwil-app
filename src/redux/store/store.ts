import { configureStore } from 'src/redux/store'

export const initialState = window && window.__INITIAL_STATE__

export const configuredStore = configureStore(initialState)
export const store = configuredStore.store
export const persistor = configuredStore.persistor
