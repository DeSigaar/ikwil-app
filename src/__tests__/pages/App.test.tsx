import React from 'react'
import renderer from 'react-test-renderer'
import { App } from 'src/pages'
// import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from 'src/redux/store'

// test('app renders somehow', () => {
//   const component = renderer.create(<App></App>)
//   const tree = component.toJSON()
//   expect(tree).toMatchSnapshot()
// })

// import React from 'react'
// import { TermsOfService } from 'src/pages'
// import { MemoryRouter } from 'react-router'

test('Should render App', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <App />
      </MemoryRouter>
    </Provider>,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
  // const { queryByLabelTest } = render(
  //   ,
  // )

  // expect(queryByLabelTest(/terms/i)).toBeTruthy()
})

// // test('renders TermsOfService', () => {
// //   expect()
// //   // const { getByText } = render(<TermsOfService />)
// //   // const title = getByText(/Terms of Service/i)
// //   // expect(title).toBeInTheDocument()
// // })
