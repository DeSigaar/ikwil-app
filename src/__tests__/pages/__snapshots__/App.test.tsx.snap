// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App 1`] = `
Array [
  <div
    className="sc-AxjAm dDQxpm"
  >
    <h1
      className="sc-AxirZ dcXYlc"
    >
      Activiteiten
    </h1>
    <div
      className="sc-fzpans jZXEVy"
      onClick={[Function]}
      size={24}
    />
  </div>,
  <div>
    <label
      className="sc-AxhCb utPZu"
    />
    <input
      className="sc-AxiKw itmHMn"
      placeholder="Zoeken..."
      type="search"
    />
    <label
      className="sc-AxhUy dFFoKG"
      onClick={[Function]}
    />
  </div>,
  <div
    className="sc-fzoLsD fqEnxe"
  >
    <div>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button>
          inloggen
        </button>
      </a>
    </div>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quam at dolor voluptates odio mollitia, aliquam totam consequatur sunt autem eaque, minima, quis cupiditate tempore quas facilis. Quisquam alias adipisci est.
    </p>
  </div>,
]
`;
